### PR TITLE
[TEST] use EXPECT_RANGE_EQ in drop_test...

### PR DESCRIPTION
**Pull request by marehr**
Part of seqan/product_backlog#83.

### CODE REVIEW

**Review by simonsasse — APPROVED** [review]
Looks good :)

File: test/unit/range/views/drop_test.cpp [hunk 1]
```diff
@@ -33,29 +32,27 @@
 template <typename adaptor_t>
 void do_test(adaptor_t const & adaptor, std::string const & vec)
 {
+    using namespace std::string_literals;
+
     // pipe notation
-    auto v = vec | adaptor(3);
-    EXPECT_EQ("bar", std::string(v));
+    EXPECT_RANGE_EQ("bar"s, vec | adaptor(3));
```
Comment (eseiler) [inline test/unit/range/views/drop_test.cpp:38]: why not `"bar"sv` ?

File: test/unit/range/views/take_until_test.cpp [hunk 2]
```diff
@@ -112,22 +104,23 @@ void do_concepts(adaptor_t && adaptor, bool const_it)
 
 TEST(view_take_until, unix_eol)
 {
-    do_test(seqan3::views::take_until, [] (char c) { return c == '\n'; }, "foo\nbar");
+    auto is_newline = [](char c){return c == '\n'; };
```
Comment (eseiler) [inline test/unit/range/views/take_until_test.cpp:107]:
```suggestion
    auto is_newline = [] (char c) { return c == '\n'; };
```

File: test/unit/range/views/take_until_test.cpp [hunk 3]
```diff
@@ -112,22 +104,23 @@ void do_concepts(adaptor_t && adaptor, bool const_it)
...
-                        | seqan3::views::to<std::string> ));
-    EXPECT_EQ("foo", v);
+    auto is_newline = [](char c){return c == '\n'; };
+    EXPECT_RANGE_EQ("foo"sv, vec | seqan3::views::take_until(is_newline));
 }
 
 TEST(view_take_until, concepts)
 {
-    auto adapt = seqan3::views::take_until([] (char c) { return c == '\n'; });
+    auto is_newline = [](char c){return c == '\n'; };
```
Comment (eseiler) [inline test/unit/range/views/take_until_test.cpp:122]:
```suggestion
    auto is_newline = [] (char c) { return c == '\n'; };
```

File: test/unit/range/views/take_until_test.cpp [hunk 4]
```diff
@@ -141,19 +134,21 @@ TEST(view_take_until, concepts)
...
 }
 
 TEST(view_take_until_or_throw, functor_fail)
 {
     std::string vec{"foo"};
-    EXPECT_THROW(std::string v = vec
-                               | seqan3::views::take_until_or_throw([] (char c) { return c == '\n'; })
-                               | seqan3::views::to<std::string>,
+
+    auto is_newline = [](char c){return c == '\n'; };
```
Comment (eseiler) [inline test/unit/range/views/take_until_test.cpp:145]:
```suggestion
    auto is_newline = [] (char c) { return c == '\n'; };
```

File: test/unit/range/views/take_until_test.cpp (hunk 3 above)
Comment (eseiler) [inline test/unit/range/views/take_until_test.cpp:122]:
```suggestion
    auto is_newline = [] (char c) { return c == '\n'; };
```

**Review by eseiler — CHANGES_REQUESTED** [review]

File: test/unit/range/views/drop_test.cpp (hunk 1 above)
Comment (marehr) [reply test/unit/range/views/drop_test.cpp:38]: I started to use it in the meantime. I can change this here too. I didn't want to go a second time through all.

File: test/unit/range/views/take_until_test.cpp [hunk 5]
```diff
@@ -141,19 +134,21 @@ TEST(view_take_until, concepts)
 
 TEST(view_take_until_or_throw, unix_eol)
 {
-    do_test(seqan3::views::take_until_or_throw, [] (char c) { return c == '\n'; }, "foo\nbar");
+    auto is_newline = [](char c){return c == '\n'; };
```
Comment (marehr) [inline test/unit/range/views/take_until_test.cpp:137]:
```suggestion
    auto is_newline = [](char c){return c == '\n'; };
```

**Review by eseiler — APPROVED** [review]
Want to rebase?

**Pull request merged**